presto-main/src/main/java/com/facebook/presto/execution/QueryState.java

```diff
@@ -23,42 +23,52 @@
     /**
      * Query has been accepted and is awaiting execution.
      */
-    QUEUED(false),
+    QUEUED(false, true),
     /**
      * Query is being planned.
      */
-    PLANNING(false),
+    PLANNING(false, true),
```

Comment:
We want the queued flag to only cover the query being in the query queue, which is the `QUEUED` state.

The queued state is special because it means the query isn't running due to a queuing or resource group policy: https://prestodb.io/docs/current/admin.html

So rather than changing the `QueryState` class, just check if `queryState == QUEUED` in `QueryStateMachine`.